ext/spl/tests/gh10925.phpt

```diff
@@ -0,0 +1,32 @@
+--TEST--
+Dynamic properties serialization for SplFixedArray should have updated properties
+--FILE--
+<?php
+#[AllowDynamicProperties]
```

Comment:
Actually am I just blind, or where are you using dynamic properties here

Reply:
No it was my fault. I originally intended to also add a test for dynamic properties but apparantly forgot about it, I added it now. Thanks for pointing this out!

Reply:
I also added a comment for the key != NULL check.